File: GhostLight/GhostLight C#/GhostLight/GhostLight/GhostLight/GhostEngine/ChangeGameStateButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CustomWindower.CoreEngine;
using GhostFinder.Interface;
using MenueSystem;
/**
 * @author Michael Letter
 */
namespace GhostFinder.GhostEngine{
    public class ChangeGameStateButton : Button {
	    protected GhostFinder gameManager = null;
        protected GhostFinder.updateState targetState = GhostFinder.updateState.MAINMENUE;
	
	    public ChangeGameStateButton(){
            base.setIdleImage((Sprite)BaseCode.activeLibrary.getResource("GhostLight/GhostLight/resources/menuArt/ButtonIdle.png"));
            base.setButtonDownImage((Sprite)BaseCode.activeLibrary.getResource("GhostLight/GhostLight/resources/menuArt/ButtonDown.png"));
            base.setMouseOverIdleImage((Sprite)BaseCode.activeLibrary.getResource("GhostLight/GhostLight/resources/menuArt/ButtonIdleMouseOver.png"));
            base.setMouseOverSelectImage((Sprite)BaseCode.activeLibrary.getResource("GhostLight/GhostLight/resources/menuArt/ButtonIdleMouseOver.png"));
            base.setSelectImage((Sprite)BaseCode.activeLibrary.getResource("GhostLight/GhostLight/resources/menuArt/ButtonIdle.png"));
            base.selectable = false;
	    }
	
	    public void setTargetState(GhostFinder manager, GhostFinder.updateState target){
		    gameManager = manager;
		    targetState = target;
	    }
	    public override void autoUpdateState(Mouse mouse){
            transitionState oldState = base.getSubState();
		    base.autoUpdateState(mouse);
		    if(base.getSubState() == transitionState.DOWN && base.getSubState() != oldState){
			    changeState();
		    }
	    }
	    public void changeState(){
		    if(gameManager != null){
			    gameManager.setUpdateState(targetState);
		    }
	    }
	    public override void setState(PanelState newState, bool setSubPanels){
		    base.setState(newState, setSubPanels);
		    if(newState == PanelState.INACTIVE){
			    setVisibility(false,false);
		    }
	    }
    }
}

File: GhostLight/GhostLight Java/GhostLight/src/GhostLight/GhostEngine/ChangeGameStateButton.java
package GhostLight.GhostEngine;

import Engine.MouseInput;
import GhostLight.GhostEngine.GhostLight.updateState;
import MenueSystem.Button;
/**
 * @author Michael Letter
 */
public class ChangeGameStateButton extends Button {
	protected GhostLight gameManager = null;
	protected updateState targetState = updateState.MAINMENUE;
	
	public ChangeGameStateButton(){
		super();
		super.setIdleImage("menuArt/ButtonIdle.png");
		super.setButtonDownImage("menuArt/ButtonIdle.png");
		super.setMouseOverIdleImage("menuArt/ButtonIdleMouseOver.png");
		super.setMouseOverSelectImage("menuArt/ButtonIdleMouseOver.png");
		super.setSelectImage("menuArt/ButtonIdle.png");
		super.selectable = false;
	}
	
	public void setTargetState(GhostLight manager, updateState target){
		gameManager = manager;
		targetState = target;
	}
	public void autoUpdateState(MouseInput mouse){
		transitionState oldState = super.getSubState();
		super.autoUpdateState(mouse);
		if(super.getSubState() == transitionState.DOWN && super.getSubState() != oldState){
			changeState();
		}
	}
	public void changeState(){
		if(gameManager != null){
			gameManager.setUpdateState(targetState);
		}
	}
	public void setState(PanelState newState, boolean setSubPanels){
		super.setState(newState, setSubPanels);
		if(newState == PanelState.INACTIVE){
			setVisibility(false,false);
		}
	}
}
